File: ide-plugins/pycharm/src/main/java/com/proboscis/pinjectdesign/settings/PInjectDesignSettings.java
package com.proboscis.pinjectdesign.settings;

import com.intellij.openapi.components.PersistentStateComponent;
import com.intellij.openapi.components.Service;
import com.intellij.openapi.components.State;
import com.intellij.openapi.components.Storage;
import com.intellij.util.xmlb.XmlSerializerUtil;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@Service
@State(
    name = "com.proboscis.pinjectdesign.settings.PInjectDesignSettings",
    storages = @Storage("PInjectDesignSettings.xml")
)
public class PInjectDesignSettings implements PersistentStateComponent<PInjectDesignSettings> {
    public boolean enableDetailedAnalysis = false;
    public boolean checkForDependencies = true;
    public boolean enableCodeCompletion = true;
    
    @Override
    public @Nullable PInjectDesignSettings getState() {
        return this;
    }
    
    @Override
    public void loadState(@NotNull PInjectDesignSettings state) {
        XmlSerializerUtil.copyBean(state, this);
    }
}

File: ide-plugins/pycharm/src/main/kotlin/com/proboscis/pinjectdesign/settings/PInjectDesignSettings.kt
package com.proboscis.pinjectdesign.settings

import com.intellij.openapi.application.ApplicationManager
import com.intellij.openapi.components.PersistentStateComponent
import com.intellij.openapi.components.Service
import com.intellij.openapi.components.State
import com.intellij.openapi.components.Storage
import com.intellij.util.xmlb.XmlSerializerUtil

@Service
@State(
    name = "PInjectDesignSettings",
    storages = [Storage("PInjectDesignSettings.xml")]
)
class PInjectDesignSettings : PersistentStateComponent<PInjectDesignSettings> {
    var enableGutterIcons: Boolean = true
    var enableVisualization: Boolean = true

    override fun getState(): PInjectDesignSettings = this

    override fun loadState(state: PInjectDesignSettings) {
        XmlSerializerUtil.copyBean(state, this)
    }

    companion object {
        fun getInstance(): PInjectDesignSettings {
            return ApplicationManager.getApplication().getService(PInjectDesignSettings::class.java)
        }
    }
}
